app/grandchallenge/discussion_forums/models.py

```diff
@@ -176,44 +176,115 @@ class Meta:
             "created",
         ]
 
-    def __str__(self):
-        return self.subject
-
     @property
     def is_alone(self):
         return self.topic.posts.count() == 1
 
+    @property
+    def is_last_post(self):
+        return self.topic.last_post == self
+
     def save(self, *args, **kwargs):
+        adding = self._state.adding
+
         super().save(*args, **kwargs)
+
+        if adding:
+            self.assign_permissions()
+
         self.topic.last_post_on = self.created
         self.topic.save()
 
     def delete(self, *args, **kwargs):
+        update_last_post_on = False
+        topic = None
+
         if self.is_alone:
             self.topic.delete()
-        else:
-            super().delete(*args, **kwargs)
+        elif self.is_last_post:
+            update_last_post_on = True
+            topic = self.topic
+
+        super().delete(*args, **kwargs)
 
+        if update_last_post_on:
+            topic.last_post_on = topic.last_post.created
+            topic.save()
 
-class ForumUserObjectPermission(NoUserPermissionsAllowed):
+    def assign_permissions(self):
+        # challenge admins and participants can see this post
+        assign_perm(
+            "discussion_forums.view_forumpost",
```

Comment:
Maybe we should prefer using just the codename (e.g. "view_forumpost", without the app name) everywhere? That would make our codebase more searchable and I do not think that anything is lost by removing the app name. 

Same applies for the permissions in the views and the change to be added to our style guide.